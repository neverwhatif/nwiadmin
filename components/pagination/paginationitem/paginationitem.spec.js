import React from 'react';
import { shallow } from 'enzyme';

import PaginationItem from './index';

describe('PaginationItem', () => {
    const component = shallow(<PaginationItem to="lorem" label={1} />);
    const componentIsCurrent = shallow(<PaginationItem to="lorem" label={2} isCurrent />);
    const componentEllipsis = shallow(<PaginationItem type="ellipsis" />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentIsCurrent).toMatchSnapshot();
        expect(componentEllipsis).toMatchSnapshot();
    });
    it('should render link', () => {
        expect(component.is('Link')).toBe(true);
    });
    it('should render current', () => {
        expect(componentIsCurrent.hasClass('rootCurrent')).toBe(true);
    });
    it('should render ellipsis', () => {
        expect(componentEllipsis.is('span')).toBe(true);
        expect(componentEllipsis.hasClass('ellipsis')).toBe(true);
    });
    it('should handle scrollToTop', () => {
        window.scroll = jest.fn();

        component.simulate('click');
        expect(window.scroll).toHaveBeenCalled();
    });
});
