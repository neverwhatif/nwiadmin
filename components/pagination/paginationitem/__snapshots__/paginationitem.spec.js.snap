// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationItem
    isCurrent={false}
    label={1}
    to="lorem"
    type="link"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": 1,
      "className": "root rootLink",
      "onClick": [Function],
      "to": "lorem",
    },
    "ref": null,
    "rendered": 1,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": 1,
        "className": "root rootLink",
        "onClick": [Function],
        "to": "lorem",
      },
      "ref": null,
      "rendered": 1,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PaginationItem should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationItem
    isCurrent={true}
    label={2}
    to="lorem"
    type="link"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": 2,
      "className": "root rootLink rootCurrent",
      "onClick": [Function],
      "to": "lorem",
    },
    "ref": null,
    "rendered": 2,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": 2,
        "className": "root rootLink rootCurrent",
        "onClick": [Function],
        "to": "lorem",
      },
      "ref": null,
      "rendered": 2,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PaginationItem should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationItem
    isCurrent={false}
    label={null}
    to={null}
    type="ellipsis"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "...",
      "className": "ellipsis",
    },
    "ref": null,
    "rendered": "...",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "...",
        "className": "ellipsis",
      },
      "ref": null,
      "rendered": "...",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
